docs/docs/guides/gitlab.mdx

```diff
@@ -0,0 +1,433 @@
+---
+id: gitlab
+title: Gitlab Hydra integration
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl'
+
+Gitlab has several OAuth2 related features. The relevant here is the possibility
+to sign into GitLab with (almost) any OAuth2 provider, in this case ORY Hydra. The documentation for this
+is kept generic on purpose. So, in this guide, we'll connect
+GitLab's omniauth-connector to ORY Hydra. We'll do that in a docker-based
+lab-environment in order to investigate the details before you do something like
+this in production.
+
+## Preparation
+
+Even though we're mostly using hydra in a docker-container, having the
+command-line-client available is quite useful. So let's install hydra as
+explained in the [installation-guide](../install).
+You'll also need [docker](https://docs.docker.com/get-docker/) and
+[docker-compose](https://docs.docker.com/compose/install/).
+
+The [hydra-5-min-tutorial](../5min-tutorial) might
+be worth checking out upfront. It'll a give a nice quick overview how OAuth2 is
+working within hydra with a minimal example. We assume basic knowledge, here.
+
+If you have not yet the source code of hydra, which we'll need for the `docker-compose`
+yaml-files, clone this repository:
+
+```shell
+git clone git@github.com:ory/hydra-gitlab-example.git
+```
+
+## Spin up the instances
+
+Gitlab has some [documentation](https://docs.gitlab.com/omnibus/docker/) about
+how to use their docker-images. It has also an example for docker-compose. So change into the directory with the Hydra 
+source code and create a new file quickstart-gitlab.yml with this content:
+
+```yaml
+version: '3'
+
+services:
+  gitlab:
+    image: gitlab/gitlab-ce:latest
+    restart: always
+    hostname: gitlab.example.com
+    environment:
+      GITLAB_OMNIBUS_CONFIG: |
+        external_url 'http://gitlab.example.com:8000/'
+    ports:
+      - "443:443" # https
+      - "8000:8000" # http
```

Comment:
Why do you want to change it to 80? This is also a privileged port. Depending on how docker is installed on a system, this might or might not be an issue. On Linux it's not a problem as the dpkg-based installation is using root-privileges. However it's a good idea to avoid issues with that.

Reply:
Sorry, I can't understand the official image “gitlab/gitlab-ce:13.0.6-ce.0” code is “ EXPOSE 22 443 80”，Why do you use 8000 to map 8000，Not 8000 mapping 80 .